### PR TITLE
Multiple assignment conditional assignment

### PR DESCRIPTION
**Pull request by rrosenblum**
This is taking me a little longer to get everything functioning how I would like it. I figured I would post my work in progress as a sign of good faith that progress is being made.

I think I have the logic mostly ironed out. As of now, I need to implement the functionality for `case when`, work on the test cases, and then go through a massive cleanup.

### CODE REVIEW

**Comment by rrosenblum** [comment]
I like that idea. I have not been overly happy with the configuration name so far. I haven't thought of anything better yet.

**Comment by rrosenblum** [comment]
I have the functionality of the configuration option down. I need to work on reducing complexity of the class.

**Comment by rrosenblum** [comment]
I finally finished up on the changes. The complexity of the configuration was greater than I expected it to be. It took a while to reduce the complexity. I had to disable `Metrics/MethodLength` for two of the methods.

@bbatsov let me know what you think of the changes.

File: config/enabled.yml [hunk 1]
```diff
@@ -159,6 +159,7 @@ Style/ConditionalAssignment:
                  assignment to a variable instead of assigning that variable
                  inside of each branch.
   Enabled: true
+  AutoCorrectConditionsWithMultipleAssignments: false
```
Comment (jonas054) [inline config/enabled.yml]: This parameter doesn't only control auto-correct, so the name is misleading.

**Comment by bbatsov** [comment]
The changes look good to me overall. This has to be rebased.

**Comment by bbatsov** [comment]
Btw, I don't see any explanation of this config option for the users. Perhaps the idea was that the name is descriptive enough.

**Comment by rrosenblum** [comment]
This rebase is tricky because so much has changed on both sides. It may take a little while to figure this out.

**Comment by alexdowad** [comment]
> This rebase is tricky because so much has changed on both sides. It may take a little while to figure this out.

Sorry @rrosenblum!

**Comment by rrosenblum** [comment]
@alexdowad it's cool. The rebase was a bit tricky before your most recent change. Your change may give me the ability to re-implement my changes in a cleaner way.

**Comment by rrosenblum** [comment]
Closing in favor of #2480

**Pull request closed without merging**